x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/single_page_layout/hooks/setup_technology.ts

```diff
@@ -92,9 +93,13 @@ export function useSetupTechnology({
 
   // this is a placeholder for the new agent-BASED policy that will be used when the user switches from agentless to agent-based and back
   const newAgentBasedPolicy = useRef<NewAgentPolicy>(newAgentPolicy);
-  const [selectedSetupTechnology, setSelectedSetupTechnology] = useState<SetupTechnology>(
-    SetupTechnology.AGENT_BASED
-  );
+  const getDefaultSetupTechnology = useCallback(() => {
```

Comment:
nit: Does this need to be a callback? it seems to me it will be easier to understand with a memoize value 

```
const defaultSetupTechnology = useMemo(() => ..., [])
```

Reply:
Totally, thanks for that. I was tempted to skip the conditional initial state and only have a `useEffect` as the component always seems to first render with `undefined` package info, but this seemed more solid.